### PR TITLE
Adds actions transformation layer

### PR DESCRIPTION
**Pull request by manueltanzi-okta**
### Description

Adds actions transformation layer

### Reviewers

@haishengwu-okta @nikhilvenkatraman-okta @robertdamphousse-okta

### CODE REVIEW

File: src/v2/ion/actionsTransformer.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import httpClient from './httpClient';
+
+const convertActionToRequestFunction = action => {
+  const request = action();
+  return extraData => httpClient.fetchRequest(request.url, request.method, Object.assign({}, request.data, extraData));
+};
+
+/**
+ *
+ * @param {AuthnResponse} resp transformed authn v2 response
```
Comment (haishengwu-okta) [inline src/v2/ion/actionsTransformer.js]: it's `AuthnResult`

File: src/v2/ion/httpClient.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+const fetchRequest = (url, method, body) => {
+  const fetchPromise = fetch(url, {
+    method,
+    headers,
+    body: JSON.stringify(body),
+    credentials: 'include'
+  })
+    .then(response => {
+      if (!response.ok) {
+        throw 'API error';
```
Comment (manueltanzi-okta) [inline src/v2/ion/httpClient.js]: note: in okta-auth-js this is more articulated and we throw https://github.com/okta/okta-auth-js/blame/master/lib/http.js#L104 , do we want to do same here?

File: test/unit/spec/v2/ion/actionsTransformer_spec.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+describe('v2/ion/actionsTransformer', function () {
+  it('returns result when invokes with invalid resp', () => {
+    expect(transformActions()).toBeNull();
+    expect(transformActions('hello')).toBeNull();
+  });
+
+  it('converts factor require email', () => {
+    let result = transform(XHRFactorRequiredEmail.response);
+    result = transformActions(result);
```
Comment (haishengwu-okta) [inline test/unit/spec/v2/ion/actionsTransformer_spec.js]: nit: https://underscorejs.org/#compose

` result = _.compose(transformActions, transform)(response)`

File: src/v2/ion/actionsTransformer.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * @returns {AuthnResult} result with transformed actions
+ */
+const transformActions = transformedResp => {
+  if (!_.isObject(transformedResp)) {
+    return null;
+  }
+
+  const transformedCurrentState = {};
+  _.each(transformedResp.currentState, (val, key) => {
```
[the remark this answers is not in the document]
Comment (manueltanzi-okta) [reply src/v2/ion/actionsTransformer.js]: Oh right, got confused by the test, will update 👍

**Pull request merged**